srml/executive/src/lib.rs

```diff
@@ -116,41 +115,45 @@ pub type CheckedOf<E, C> = <E as Checkable<C>>::Checked;
 pub type CallOf<E, C> = <CheckedOf<E, C> as Applyable>::Call;
 pub type OriginOf<E, C> = <CallOf<E, C> as Dispatchable>::Origin;
 
-pub struct Executive<System, Block, Context, Payment, UnsignedValidator, AllModules>(
-	PhantomData<(System, Block, Context, Payment, UnsignedValidator, AllModules)>
+pub struct Executive<System, Block, Context, Payment, Balance, UnsignedValidator, AllModules>(
```

Comment:
Why not re-use `Payment`?

Reply:
If you mean use `Payment` as a means of passing in the `Balance` type: 

Can't be trivially done since now the `make_payment` itself is dependent on the balance. Otherwise yeah could have just said sth like: 

`where Payment: MakePayment<System::AccountId> + balance::Trait`;

To my understanding, this would make things more complicated as long as the `Tip` has its balance as generic.

Reply:
We could make `Balance` and associated type of `MakePayment` and just reference this?